ts/Core/Chart/Chart.ts

```diff
@@ -2324,26 +2324,35 @@ class Chart {
 
         // Apply new links
         chartSeries.forEach(function (series): void {
-            let linkedTo = series.options.linkedTo;
+            const { linkedTo } = series.options;
 
-            if (isString(linkedTo)) {
+            if (linkedTo && isString(linkedTo)) {
+                let seriesLinkedTo: Series | undefined;
                 if (linkedTo === ':previous') {
-                    linkedTo = chart.series[(series.index as any) - 1] as any;
+                    seriesLinkedTo = chart.series[series.index - 1];
                 } else {
-                    linkedTo = chart.get(linkedTo as any) as any;
+                    seriesLinkedTo = chart.get(linkedTo) as Series | undefined;
                 }
                 // #3341 avoid mutual linking
-                if (linkedTo && (linkedTo as any).linkedParent !== series) {
-                    (linkedTo as any).linkedSeries.push(series);
-                    series.linkedParent = linkedTo as any;
+                if (
+                    seriesLinkedTo &&
+                    seriesLinkedTo.linkedParent !== series) {
+                    seriesLinkedTo.linkedSeries.push(series);
+                    /**
+                     * Parent series of the current series.
+                     *
+                     * @name Highcharts.Series#linkedParent
+                     * @type {Highcharts.Series}
+                     */
```

Comment:
Looks like this  needs an `@apioption` tag to show up on api.highcharts.com (`npm run jsdoc` to test).

I also feel like this should be in `Series.ts`,  but perhaps there's a reason it's here?

Reply:
Hmm, it is working correctly for me: http://localhost:9005/class-reference/Highcharts.Series#linkedParent It should show in the class members part of the documenation because it is not an API option, but just a class member.

Reply:
When it comes to a place of those lines, I followed the `series.index` property, described here: 
https://github.com/highcharts/highcharts/blob/d3aaa01d7fd79083d503ff2041f70c945e2821f5/ts/Core/Chart/Chart.ts#L690-L697

And here: https://api.highcharts.com/class-reference/Highcharts.Series#index

It seems to me, that the description of that kind of property is before first usage of it.

Reply:
Ah, ok. If that's the precedent we should probably stick with that.